oracle/oracle/common/ipfs.py

```diff
@@ -17,39 +18,51 @@
 
 timeout = ClientTimeout(total=60)
 
+CACHE_SIZE = 1024
+IPFS_CACHE = LimitedSizeDict(size_limit=CACHE_SIZE)
+
 
 @backoff.on_exception(backoff.expo, Exception, max_time=900)
 async def ipfs_fetch(ipfs_hash: str) -> Union[Dict[Any, Any], List[Dict[Any, Any]]]:
```

Comment:
Why not use something like [cache](https://docs.python.org/3/library/functools.html#functools.cache)

Reply:
it's not tread-safe function